swap/src/asb/event_loop.rs

```diff
@@ -384,15 +383,34 @@ where
                                 tracing::warn!(
                                     swap_id = %swap_id,
                                     reason = "swap is in invalid state",
-                                    "Rejecting cooperative XMR redeem request"
+                                    "Rejecting cooperative Monero redeem request"
                                 );
                                 if self.swarm.behaviour_mut().cooperative_xmr_redeem.send_response(channel, Rejected { swap_id, reason: CooperativeXmrRedeemRejectReason::SwapInvalidState }).is_err() {
-                                    tracing::error!(swap_id = %swap_id, "Failed to reject cooperative XMR redeem request");
+                                    tracing::error!(swap_id = %swap_id, "Failed to send rejection for cooperative Monero redeem request");
                                 }
                                 continue;
                             };
 
-                            if self.swarm.behaviour_mut().cooperative_xmr_redeem.send_response(channel, Fullfilled { swap_id, s_a: state3.s_a }).is_err() {
+                            // Retrieve the monero transfer proof from an earlier state
+                            let Ok(states) = self.db.get_states(swap_id).await else {
```

Comment:
We only ever go into `BtcPunished` if we locked our Monero. This means we should have a transfer proof in this state. We shouldn't iterate through older states to find data. Anti pattern. Instead move the data into the `BtcPunished` to ensure we have it availiable.